frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -138,34 +130,66 @@ end
 
 -- Generate "Reset" file dialog button
 function filemanagerutil.genResetSettingsButton(file, caller_callback, button_disabled)
+    file = ffiutil.realpath(file) or file
+    local has_sidecar_file = DocSettings:hasSidecarFile(file)
+    local custom_cover_file = DocSettings:findCoverFile(file)
+    local has_custom_cover_file = custom_cover_file and true or false
+    local custom_metadata_file = DocSettings:getCustomMetadataFile(file)
+    local has_custom_metadata_file = custom_metadata_file and true or false
     return {
         text = _("Reset"),
         id = "reset", -- used by covermenu
-        enabled = (not button_disabled and DocSettings:hasSidecarFile(ffiutil.realpath(file))) and true or false,
+        enabled = not button_disabled and (has_sidecar_file or has_custom_metadata_file or has_custom_cover_file),
         callback = function()
+            local CheckButton = require("ui/widget/checkbutton")
             local ConfirmBox = require("ui/widget/confirmbox")
+            local check_button_settings, check_button_cover, check_button_metadata
             local confirmbox = ConfirmBox:new{
                 text = T(_("Reset this document?") .. "\n\n%1\n\n" ..
-                    _("Document progress, settings, bookmarks, highlights, notes and custom cover image will be permanently lost."),
+                         _("Resetted information will be permanently lost."),
```

Comment:
`Reset`, `set` is an irregular verb (set/set/set ;)).

Which possibly makes this turn of phrase clunky, so I'd drop it entirely, e.g., `Information will be permanently lost`.

Reply:
I believed so as well but googled and got
https://en.wiktionary.org/wiki/resetted
https://www.wordsense.eu/resetted/

Reply:
> I believed so as well but googled and got https://en.wiktionary.org/wiki/resetted https://www.wordsense.eu/resetted/

Huh. That's another meaning entirely (and a chiefly local on at that), but a fun quirk I wasn't aware of nonetheless ;).